pkgs/by-name/ni/niri/package.nix

```diff
@@ -26,10 +29,16 @@ rustPlatform.buildRustPackage rec {
   src = fetchFromGitHub {
     owner = "YaLTeR";
     repo = "niri";
-    rev = "v${version}";
+    rev = "refs/tags/v${version}";
```

Comment:
What's the point of this change? Is it documented anywhere that this is "best practice" for nixpkgs? As far as i know, this is always exactly equivalent. It's just a longer way to express the exact same idea. So why do it? At the same time, i know this is common to see in nixpkgs. So, "why do it?" is a genuine question here, it's not meant to imply a pessimistic answer.

Reply:
> What's the point of this change? Is it documented anywhere that this is "best practice" for nixpkgs?

It's documented as the example reference for tags [here](https://github.com/NixOS/nixpkgs/blob/71a33cc7af3a199871b8f04e5f1e991e9ca11267/doc/build-helpers/fetchers.chapter.md?plain=1#L756)

> As far as i know, this is always exactly equivalent

https://github.com/NixOS/nixpkgs/issues/162454

Reply:
Oh, ouch, yeah that's a weird edge case in the linked issue. Unlikely to ever occur with niri, but i agree it's then nicer to just not ever have to think about it and use the more precise syntax for all packages in general, including niri.